interfaces/prompting/requestrules/requestrules.go

```diff
@@ -1361,8 +1391,13 @@ func (rdb *RuleDB) PatchRule(user uint32, id prompting.IDType, constraintsPatch
 	// support patching it? Currently, we don't include fully expired rules
 	// in the output of Rules(), should the same be done here?
 
+	currSession, err := readOrAssignUserSessionID(rdb, user)
+	if err != nil && !errors.Is(err, errNoUserSession) {
+		return nil, err
+	}
 	at := prompting.At{
-		Time: time.Now(),
+		Time:      time.Now(),
+		SessionID: currSession,
```

Comment:
in which scenarios is the new session incorporated in the Rule? only if the constraintsPatch mentions the lifespan or anyway?

Reply:
can you add a comment about the relevance/behavior of curSession in PatchRule